packages/core/src/loaders/geometryLoader.ts

```diff
@@ -6,31 +6,156 @@ import type {
- * for .vti, .vtk, .obj, .dat etc. can be created and registered here.
+ * Load a geometry using a registered Cornerstone Geometry Loader.
+ *
+ * The geometry loader that is used will be
+ * determined by the geometry loader scheme matching against the geometryId.
+ *
+ * @param geometryId - A Cornerstone Geometry Object's geometryId
+ * @param options - Options to be passed to the Geometry Loader.
+ *
+ * @returns An Object which can be used to act after a geometry is loaded or loading fails
+ *
  */
+function loadGeometryFromGeometryLoader(
+  geometryId: string,
+  options?: GeometryOptions
+): IGeometryLoadObject {
+  const colonIndex = geometryId.indexOf(':');
+  const scheme = geometryId.substring(0, colonIndex);
+  let loader = geometryLoaders[scheme];
+
+  if (loader === undefined || loader === null) {
+    if (
+      unknownGeometryLoader == null ||
+      typeof unknownGeometryLoader !== 'function'
+    ) {
+      throw new Error(
+        `No geometry loader for scheme ${scheme} has been registered`
+      );
+    }
+
+    loader = unknownGeometryLoader;
+  }
+
+  const geometryLoadObject = loader(geometryId, options);
+
+  // Broadcast a geometry loaded event once the geometry is loaded
+  geometryLoadObject.promise.then(
+    function (geometry) {
+      triggerEvent(eventTarget, Events.GEOMETRY_LOADED, { geometry });
+    },
+    function (error) {
+      const errorObject = {
+        geometryId,
+        error,
+      };
+
+      triggerEvent(eventTarget, Events.GEOMETRY_LOADED_FAILED, errorObject);
+    }
+  );
+
+  return geometryLoadObject;
+}
 
 /**
- * It creates a geometry object and caches it
- * @param geometryId - A unique identifier for the geometry.
- * @param options - GeometryOptions
- * @returns A promise that resolves to a geometry object.
+ * Loads a geometry given a geometryId and optional options and returns a promise which will resolve to
+ * the loaded geometry object or fail if an error occurred. The loaded geometry is not stored in the cache.
+ *
+ * @param geometryId - A Cornerstone Geometry Object's geometryId
+ * @param options - Options to be passed to the Geometry Loader
+ *
+ * @returns An Object which can be used to act after a geometry is loaded or loading fails
  */
-async function createAndCacheGeometry(
+export function loadGeometry(
   geometryId: string,
-  options: GeometryOptions
+  options?: GeometryOptions
+): Promise<IGeometry> {
+  if (geometryId === undefined) {
+    throw new Error('loadGeometry: parameter geometryId must not be undefined');
+  }
+
+  let geometryLoadObject = cache.getGeometryLoadObject(geometryId);
+
+  if (geometryLoadObject !== undefined) {
+    return geometryLoadObject.promise;
+  }
+
+  geometryLoadObject = loadGeometryFromGeometryLoader(geometryId, options);
+
+  return geometryLoadObject.promise;
+}
+
+/**
+ * Loads a geometry given a geometryId and optional options and returns a promise which will resolve to
```

Comment:
This function and the loadGeometric from the outside seem to do the same sort of thing - can you add notes as to why this one is called createAndCache?  Maybe it should be loadAndCache?

Reply:
We don't have a loader yet for geometry. In the future, we can write loaders for STL and other relevant data types. For now, this is points and polys, so I think create and cache is relevant. I updated the doc.

Reply:
Ok done